drivers/scheduler/k8s/k8s.go

```diff
@@ -1928,8 +1928,12 @@ func (k *K8s) createStorageObject(spec interface{}, ns *corev1.Namespace, app *s
 	if obj, ok := spec.(*storageapi.StorageClass); ok {
 		obj.Namespace = ns.Name
 
+		// volume.GetStorageProvisioner() returns the corresponding value from the portworx.provisioners map
+		// based on the --provisioner flag, such as "kubernetes.io/portworx-volume", "pxd.portworx.com", or "strict".
 		if volume.GetStorageProvisioner() != PortworxStrict {
-			if options.StorageProvisioner == string(volume.DefaultStorageProvisioner) || options.StorageProvisioner == CsiProvisioner {
```

Comment:
We were comparing options.StorageProvisioner (csi) to the constant variable CsiProvisioner (pxd.portworx.com).